tests/pylint/runpylint.py

```diff
@@ -68,18 +68,26 @@ def _get_py_paths(self, directory):
                 continue
 
             for f in files:
+                file_path = root + "/" + f
                 try:
-                    with open(root + "/" + f) as fo:
+                    if os.path.getsize(file_path) > 10e6:
+                        # Skip too large files - 10 MB should be enough for everyone!
+                        continue
+                    with open(file_path) as fo:
                         lines = fo.readlines()
```

Comment:
Weird. Why do we read all lines since we later work with only the first one? I think we should fix that instead of the checking the size of the file.

Reply:
I agree, and have some ideas, so I'll take a look. Let's stay mindful of scope creep, though. I want to ignore some new stuff, not be forced to rewrite the whole thing because you don't like it.

Reply:
Definitely. However, I suggested it as an alternative to the `os.path.getsize(file_path) > 10e6` check. It seems better to fix the original issue then to add a workaround for it.

Reply:
This seems to work.

```
     def _get_py_paths(self, directory):
@@ -70,7 +69,7 @@ class AnacondaLintConfig(CensorshipConfig):
             for f in files:
                 try:
                     with open(root + "/" + f) as fo:
-                        lines = fo.readlines()
+                        line = fo.readline()
                 except UnicodeDecodeError:
                     # If we couldn't open this file, just skip it.  It wasn't
                     # going to be valid python anyway.
@@ -78,7 +77,7 @@ class AnacondaLintConfig(CensorshipConfig):
 
                 # Test any file that either ends in .py or contains #!/usr/bin/python
                 # in the first line.
-                if f.endswith(".py") or (lines and str(lines[0]).startswith("#!/usr/bin/python")):
+                if f.endswith(".py") or (line and str(line).startswith("#!/usr/bin/python")):
                     retval.append(root + "/" + f)
 
         return retval
```

Reply:
Thank you! I missed that method. That will work, as it has also a data limit parameter. I was looking for a way to limit the amount of data read, in case you have an ISO file and read it until first line end...